datasets/plugins/batch/batch_flow_dataset_plugin.py

```diff
@@ -34,5 +34,24 @@ def __init__(
             run_id=run_id,
             attribute_name=attribute_name if attribute_name else (name if name else self.dataset_name),
         )
-        #
+        # The program name is that of the original dataset name
         self.program_name = dataset.program_name
+
+
+def _get_run_id(flow_name: str, run_id: Optional[str]) -> Tuple["Run", Optional[str]]:
+    from metaflow import Flow, Run
+
+    flow = Flow(flow_name)
+    run: Run
+    ret_run_id: Optional[str]
+    if run_id in ["latest_successful_run", "latest_run"]:
+        run = flow.latest_run if run_id == "latest_run" else flow.latest_successful_run
```

Comment:
a bit out of the scope of the MR 😅 but is flow.latest_run execution date aware?

Reply:
How would `flow.latest_run` behave if it were date aware?
`flow.latest_run.created_at`  and `flow.latest_run.finished_at` exist as date_time stamps for the `Run`

Reply:
sorry, I meant context date rather than execution date 😬  so does it know which dated data it used? I believe that it is namespace aware and that for s3 data the context date would likely be in the path. But was curious in like a backtesting or backfilling scenario where you are running against multiple dates at once would it be aware of which run_id was the latest by context date?